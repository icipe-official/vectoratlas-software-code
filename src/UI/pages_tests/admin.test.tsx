import React from 'react';
import AdminPage from '../pages/admin';
import { render } from '../test_config/render';

jest.mock(
  '../components/shared/AuthWrapper',
  () =>
    function MockAuthWrapper({
      role,
      children,
    }: {
      role: string;
      children: any;
    }) {
      return <div data-testid={`auth=${role}`}>{children}</div>;
    }
);

jest.mock('../components/admin/userRoles', () => ({
  UserRolePanel: () => <div>User role panel mock</div>,
}));

describe('admin page', () => {
  it('renders correctly', () => {
    const { wrapper } = render(<AdminPage />);

    expect(wrapper.container).toMatchSnapshot();
  });
});
