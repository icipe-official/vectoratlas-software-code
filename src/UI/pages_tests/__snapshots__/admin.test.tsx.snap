// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin page renders correctly 1`] = `
<div>
  <div>
    <main>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          data-testid="auth=admin"
        >
          <div>
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1c8tpg7-MuiTypography-root"
              style="flex-grow: 1;"
            >
              Administration
            </h4>
            <div>
              User role panel mock
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
